### ISSUE TITLE
How do I scale the rendered equation to match the surrounding text?

### ISSUE THREAD

**Issue by OskarPersson** [colinc86/LaTeXSwiftUI#23]
Changing the font size doesn't affect the rendered equation:

```swift
import SwiftUI
import LaTeXSwiftUI

@main
struct iOSApp: App {
    var body: some Scene {
        WindowGroup {
            LaTeX("Solve the following: $(9-2)^2\\cdot9^2$")
            LaTeX("Solve the following: $(9-2)^2\\cdot9^2$").font(.system(size: 8))
        }
    }
}
```

![Simulator Screenshot - iPhone 14 Pro - 2023-08-29 at 20 50 10](https://github.com/colinc86/LaTeXSwiftUI/assets/935414/cc99777c-74d6-4c09-8c30-002fb4b1b2b2)

**Comment by rrrpdx** [comment]
Did you have any luck with this? I was able to change the font size within latex by, for example, doing $\\Huge\\pi$ ... I tried  $\\fontsize{30}{36}\\pi$, but that didnt' seem to work.

**Comment by colinc86** [comment]
The view needs to be able to measure/calculate the current font's x-height to correctly size the characters inside of the rendered LaTeX SVG. To do that, the view must use the `UIFont`/`NSFont` classes and do its best to convert SwiftUI's `Font` structure into the correct `UIFont` instance. Right now the only way I know how to do that is by limiting the functionality to SwiftUI's static `largeTitle`, `title`, `title1`, `headline`, etc fonts.

Long story short - if you use the statically declared fonts and not the `.system(...)` font initializers, then the rendered equation text should match.

I'm going to leave this issue open though because I think it's still worth investigating to come up with a better solution.

**Comment by colinc86** [comment]
Been a while, but I've added some features that will be helpful in PR #64 - you can now set the font using either 
- SwiftUI preferred fonts (`largeTitle`, `body`, etc) or
- `UIFont`/`NSFont`

So in the example you provided you can now do something like

```
LaTeX("Solve the following: $(9-2)^2\\cdot9^2$")
  .font(UIFont.systemFont(ofSize: 8))
```

and it should render the way you'd expect.

Unfortunately, this is the best we can do until SwiftUI exposes more attributes through some Font reader to glean x-height.

This will go out on the next release.

**Comment by AstroWLAN** [comment]
This feature isn’t supported on macOS yet, is it?

**Comment by colinc86** [comment]
@AstroWLAN it's not in `main` yet but in `develop`. I've been toying around with some other methods, but I think develop is the best it will get for now. I'll create a release tonight.
